spec/models/signup_spec.rb

```diff
@@ -0,0 +1,105 @@
+require 'rails_helper'
+
+describe Signup do
+  before do
+    @params = { user: {first_name: "Joe",
+                      last_name: "Doe",
+                      email: "example@example.com",
+                      password: "password",
+                      password_confirmation: "password",
+                      company: { company_name: "my company"}}}
+  end
+  context 'is valid' do
+    before do
+      @signup = Signup.new(@params)
+    end
+
+    it 'exist' do
+      expect(@signup).to be_truthy
+    end
+
+    it 'can be saved' do
+      expect(@signup).to respond_to(:save)
+      expect(@signup.save).to be_truthy
+    end
+  end
+
+  context 'is invalid' do
+    context 'arguments are blank' do
+      before do
+        params = { user: {first_name: "",
+                          last_name: "",
+                          email: "",
+                          password: "",
+                          password_confirmation: "",
+                          company: { company_name: ""}}}
+        @signup = Signup.new(params)
+      end
+
+      it "can't be saved" do
+        expect(@signup).to respond_to(:save)
+        expect(@signup.save).to be_falsy
+      end
+
+      it 'has right errors' do
+        @signup.valid?
+        expect(@signup.errors.full_messages).to include("First name can't be blank",
+                                                        "Last name can't be blank",
+                                                        "Company name can't be blank",
+                                                        "Email can't be blank",
+                                                        "Password can't be blank",
+                                                        "Password is too short (minimum is 8 characters)")
+      end
+    end
+
+    context 'password is too short' do
+      before do
+        @params[:user][:password] = "pass"
+        @params[:user][:password_confirmation] = "pass"
+        @signup = Signup.new(@params)
+      end
+
+      it "can't be saved" do
+        expect(@signup).to respond_to(:save)
+        expect(@signup.save).to be_falsy
+      end
+
+      it 'has right error' do
+        @signup.valid?
+        expect(@signup.errors.full_messages).to include("Password is too short (minimum is 8 characters)")
+      end
+    end
+    context 'email has already been taken' do
+      before do
+        @user = create(:user, email: "example@example.com")
+        @signup = Signup.new(@params)
+      end
+
+      it "can't be saved" do
+        expect(@signup).to respond_to(:save)
+        expect(@signup.save).to be_falsy
```

Comment:
This tests method save to return false in this particuar case. Let's say we override ```save``` method to save the object but return false anyway (for whatever reason). I think it's better here to test the validity of the model like expect(@signup).to be_invalid